library/java/io/envoyproxy/envoymobile/engine/JniLibrary.java

```diff
@@ -4,15 +4,23 @@
 
 import java.nio.ByteBuffer;
 
-class JniLibrary {
+public class JniLibrary {
 
-  private static final String ENVOY_JNI = "envoy_jni";
+  private static String envoyLibraryName = "envoy_jni";
 
   // Internal reference to helper object used to load and initialize the native
   // library.
   // Volatile to ensure double-checked locking works correctly.
   private static volatile JavaLoader loader = null;
 
+  // Load test libraries based on the jvm_flag `envoy_jni_library_name`.
+  // WARNING: This should only be used for testing.
+  public static void loadTestLibrary() {
+    if (System.getProperty("envoy_jni_library_name") != null) {
+      envoyLibraryName = System.getProperty("envoy_jni_library_name");
+    }
+  }
```

Comment:
This is a compromise. I am going to figure out how to move this somewhere else so it's not shipped with our library.